assets/schema_input.json

```diff
@@ -9,8 +9,15 @@
         "properties": {
             "sample": {
                 "type": "string",
-                "pattern": "^\\S+$",
-                "errorMessage": "Sample name must be provided and cannot contain spaces",
+                "pattern": "^[^\\s\\.]+$",
+                "meta": ["external_id"],
+                "unique": true,
+                "errorMessage": "This field cannot contain .iridanext_output."
+            },
+            "sample_name": {
+                "type": "string",
+                "pattern": "^[^\\.]\\S+$",
```

Comment:
I think you should remove any restrictions on `sample_name` and instead follow a similar pattern as this block of code to replace any restricted characters with underscores `_`:

https://github.com/phac-nml/snvphylnfc/blob/f1e5fae76af276acf0a8c98174978cb21ca5d7e0/workflows/snvphylnfc.nf#L98-L109

The reason being that restricting `sample_name` means that mikrokondo will fail to run sample names don't match the above pattern (and spaces and periods are allowed in sample names in IRIDA Next). Allowing all patterns through but cleaning them up in the workflow code means that mikrokondo will still run for samples with any name.

Reply:
I added this check here to verify that  only the sample name does not start with a period as there are issues with nf-prov later on when it aggregates files for the providence reports **so here was my intention**.

But with your clarification above about `sample_name` vs `sample`   I think I can revert this cahnge.

Reply:
fixed here: https://github.com/phac-nml/mikrokondo/pull/140/commits/71260a99ab1334d0ffcafb008e673dce804c973a